sidecars/python/etc/entrypoint.sh

```diff
@@ -28,4 +28,13 @@ if [ "${USER_ID}" -ne 0 ] && command -v sudo >/dev/null 2>&1 && sudo -n true > /
     sudo chown "${USER_ID}:${GROUP_ID}" /projects
 fi
 
+# Setup .venv in the 'venv' volume that should be mounted in HOME/.venv
+cd "${HOME}"
+python -m venv .venv
+# shellcheck disable=SC1091
+. .venv/bin/activate
+# Setup jupyter dependencies in this venv
+pip install -U ipykernel jupyter
```

Comment:
Does this need to be here? Or could it be in the project's `requirements.txt` file?

Reply:
Looks like projects using jupyter notebook are not adding these dependencies in `requirements.txt`. As the python extension provide the jupyter notebook feature, we are installing these dependencies so it works out of the box as well as providing a smooth user experience.

Reply:
If this is the case maybe we should install jupyter in the Dockerfile for the python sidecar, instead of in the entrypoint?

Reply:
it doesn't work because the dependencies installed won't be available in the venv. It failed when I tried. It is done in the entrypoint so we can perform the venv setup in the volume (that is available at runtime).

Reply:
Fine by me then :+1:

Reply:
I think doing post-install step in the entrypoint.sh is not a viable solution

Reply:
@benoitf why not? what is your proposal?  having that we may have additional dependencies coming in the same venv from a projects

Reply:
I mean, the goal is to have the containers starting as fast as possible.

doing that in the entrypoint is delaying the startup of the workspace and it can even fail to start the workspace if we're unable to install the dependencies, etc.


For example in maven case, it's vscode-java that pull maven dependencies but after the workspace is started, not in the entrypoint.

Reply:
It also just occurred to me that we need to think about restricted network/self-signed certificate environments: https://www.eclipse.org/che/docs/che-7/end-user-guide/using-python-artifact-repositories/

I think the current approach will break that

Reply:
a container is considered 'started' before the entrypoint is executed. otherwise with `sleep infinity` we would never have any container `started`.

Reply:
@ericwill it should work, but it would be worth testing

Reply:
I can see how I can make it more robust, and not failing if we have network issues.

Reply:
ok i will try https://stackoverflow.com/questions/7438681/how-to-duplicate-virtualenv

Reply:
or just do the venv setup in the dockerfile and make a copy in the entrypoint

Reply:
yes, being able to just copy pre-pulled environment would be better option